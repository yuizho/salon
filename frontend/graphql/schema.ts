/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type AddOperation = {
  __typename: "AddOperation",
  room_id: string,
  user_id: string,
  user_token: string,
};

export type UpdateOperation = {
  __typename: "UpdateOperation",
  room_id: string,
  user_id: string,
};

export enum Status {
  CHOOSING = "CHOOSING",
  CHOSEN = "CHOSEN",
  LEAVED = "LEAVED",
}


export type User = {
  __typename: "User",
  room_id: string,
  user_id: string,
  status: Status,
  picked_card?: string | null,
  operated_at: string,
};

export type Room = {
  __typename: "Room",
  items:  Array<User >,
  expiration_unix_timestamp: number,
  is_opened: boolean,
};

export type OpenRoomMutation = {
  openRoom:  {
    __typename: "AddOperation",
    room_id: string,
    user_id: string,
    user_token: string,
  },
};

export type JoinMutationVariables = {
  room_id: string,
};

export type JoinMutation = {
  join:  {
    __typename: "AddOperation",
    room_id: string,
    user_id: string,
    user_token: string,
  },
};

export type LeaveMutationVariables = {
  room_id: string,
  user_id: string,
  user_token: string,
};

export type LeaveMutation = {
  leave:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type PickMutationVariables = {
  room_id: string,
  user_id: string,
  picked_card: string,
  user_token: string,
};

export type PickMutation = {
  pick:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type RefreshTableMutationVariables = {
  room_id: string,
  user_id: string,
  user_token: string,
};

export type RefreshTableMutation = {
  refreshTable:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type KickMutationVariables = {
  room_id: string,
  user_id: string,
  kicked_user_id: string,
  user_token: string,
};

export type KickMutation = {
  kick:  {
    __typename: "UpdateOperation",
    room_id: string,
    user_id: string,
  },
};

export type UpdateUserMutationVariables = {
  room_id: string,
  user_id: string,
  status: Status,
  picked_card?: string | null,
  operated_at: string,
};

export type UpdateUserMutation = {
  updateUser:  {
    __typename: "User",
    room_id: string,
    user_id: string,
    status: Status,
    picked_card?: string | null,
    operated_at: string,
  },
};

export type GetRoomQueryVariables = {
  room_id: string,
};

export type GetRoomQuery = {
  getRoom:  {
    __typename: "Room",
    items:  Array< {
      __typename: "User",
      room_id: string,
      user_id: string,
      status: Status,
      picked_card?: string | null,
      operated_at: string,
    } >,
    expiration_unix_timestamp: number,
    is_opened: boolean,
  },
};

export type OnUpdateUserSubscriptionVariables = {
  room_id: string,
};

export type OnUpdateUserSubscription = {
  onUpdateUser?:  {
    __typename: "User",
    room_id: string,
    user_id: string,
    status: Status,
    picked_card?: string | null,
    operated_at: string,
  } | null,
};
