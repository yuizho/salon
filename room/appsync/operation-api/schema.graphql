type Query {
  getOperation(event_id: String!): Operation
}

type Mutation {
  openRoom: Operation
  join(room_id: String!): Operation
  leave(room_id: String!, user_id: String!): Operation
  pick(room_id: String!, user_id: String!, picked_card: String!): Operation
  refreshTable(room_id: String!, user_id: String!): Operation
}

type Operation {
  event_id: String
  room_id: String
  user_id: String
  op_type: String
  operated_at: String
  picked_card: String
}
