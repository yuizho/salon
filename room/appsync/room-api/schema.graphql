type Mutation {
  openRoom: Operation @aws_api_key
  join(room_id: String!): Operation @aws_api_key
  leave(room_id: String!, user_id: String!): Operation @aws_api_key
  pick(room_id: String!, user_id: String!, picked_card: String!): Operation
    @aws_api_key
  refreshTable(room_id: String!, user_id: String!): Operation @aws_api_key
  heartbeat(room_id: String!, user_id: String!): Operation @aws_api_key
  updatePoker(
    room_id: String!
    user_id: String!
    status: Status!
    picked_card: String
    operated_at: AWSDateTime!
    joined_at: AWSDateTime!
  ): Poker @aws_iam
}

type Query {
  getRoom(room_id: String!): Room @aws_api_key
}

type Subscription {
  onUpdatePoker(room_id: String): Poker
    @aws_subscribe(mutations: ["updatePoker"])
    @aws_api_key
}

type Operation @aws_api_key {
  event_id: String
  room_id: String
  user_id: String
  op_type: OperationType
  operated_at: String
  picked_card: String
}

enum OperationType {
  OPEN_ROOM
  JOIN
  LEAVE
  PICK
  REFRESH_TABLE
  HEARTBEAT
}

type Room @aws_api_key @aws_iam {
  items: [Poker]
}

type Poker @aws_api_key @aws_iam {
  room_id: String!
  user_id: String!
  status: Status!
  picked_card: String
  operated_at: AWSDateTime!
  joined_at: AWSDateTime!
}

enum Status {
  CHOOSING
  CHOSEN
  LEAVED
}
