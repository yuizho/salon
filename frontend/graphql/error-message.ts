// eslint-disable-next-line import/prefer-default-export
export const NETWORK_ERROR =
  '通信エラーが発生しました。ネットワークの状態を確認し、解決しない場合は時間をおいてからやり直してください。';
