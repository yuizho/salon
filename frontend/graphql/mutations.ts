/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const openRoom = /* GraphQL */ `
  mutation OpenRoom {
    openRoom {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const join = /* GraphQL */ `
  mutation Join($room_id: String!) {
    join(room_id: $room_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const leave = /* GraphQL */ `
  mutation Leave($room_id: String!, $user_id: String!) {
    leave(room_id: $room_id, user_id: $user_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const pick = /* GraphQL */ `
  mutation Pick($room_id: String!, $user_id: String!, $picked_card: String!) {
    pick(room_id: $room_id, user_id: $user_id, picked_card: $picked_card) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const refreshTable = /* GraphQL */ `
  mutation RefreshTable($room_id: String!, $user_id: String!) {
    refreshTable(room_id: $room_id, user_id: $user_id) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const kick = /* GraphQL */ `
  mutation Kick(
    $room_id: String!
    $user_id: String!
    $kicked_user_id: String!
  ) {
    kick(
      room_id: $room_id
      user_id: $user_id
      kicked_user_id: $kicked_user_id
    ) {
      event_id
      room_id
      user_id
      op_type
      operated_at
      picked_card
      kicked_user_id
    }
  }
`;
export const updateUser = /* GraphQL */ `
  mutation UpdateUser(
    $room_id: String!
    $user_id: String!
    $status: Status!
    $picked_card: String
    $operated_at: AWSDateTime!
  ) {
    updateUser(
      room_id: $room_id
      user_id: $user_id
      status: $status
      picked_card: $picked_card
      operated_at: $operated_at
    ) {
      room_id
      user_id
      status
      picked_card
      operated_at
    }
  }
`;
