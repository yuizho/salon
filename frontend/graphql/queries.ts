/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const getRoom = /* GraphQL */ `
  query GetRoom($room_id: String!) {
    getRoom(room_id: $room_id) {
      items {
        room_id
        user_id
        status
        picked_card
        operated_at
      }
      expiration_unix_timestamp
      is_opened
    }
  }
`;
