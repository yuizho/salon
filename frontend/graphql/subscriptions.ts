/* tslint:disable */
/* eslint-disable */
// this is an auto generated file. This will be overwritten

export const onUpdateUser = /* GraphQL */ `
  subscription OnUpdateUser($room_id: String!) {
    onUpdateUser(room_id: $room_id) {
      room_id
      user_id
      status
      picked_card
      operated_at
    }
  }
`;
