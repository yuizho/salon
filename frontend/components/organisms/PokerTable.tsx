import { FC, useState } from 'react';
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
import API, { GraphQLResult } from '@aws-amplify/api';
import { RefreshTableMutation, RefreshTableMutationVariables, Status } from '../../graphql/schema';
import { Poker, PokerState, pokerState as pokerRecoilState } from '../../states/poker';
import { User, usersState } from '../../states/users';
import { Me, myState } from '../../states/me';
import Button from '../atoms/Button';

import Players from './Players';
import { refreshTable } from '../../graphql/mutations';
import ModalDialog from '../molecules/ModalDialog';
import Message from '../atoms/Message';
import { appState } from '../../states/app';
import { NETWORK_ERROR } from '../../graphql/error-message';

type ComponentProps = {
  me: Me;
  users: Array<User>;
  poker: Poker;
  onReset: () => void;
  openResetDialog: boolean;
  setOpenResetDialog: (b: boolean) => void;
  pokerState: PokerState;
};

const mutateRefreshTable = async (roomId: string, userId: string) => {
  const result = (await API.graphql({
    query: refreshTable,
    variables: {
      room_id: roomId,
      user_id: userId,
    } as RefreshTableMutationVariables,
  })) as GraphQLResult<RefreshTableMutation>;
  return result;
};

const getMessageByPokerState = (state: PokerState) => {
  if (state === 'KICKED') {
    return '他のユーザによって部屋からキックされました。再度入室する場合はブラウザを更新してください。';
  }
  return '';
};

export const Component: FC<ComponentProps> = ({
  me,
  users,
  poker,
  onReset,
  openResetDialog,
  setOpenResetDialog,
  pokerState,
}) => {
  const message = getMessageByPokerState(poker.state);

  return (
    <div
      className={`flex flex-col space-y-6 border rounded p-8 ${
        pokerState === 'LOADING' && 'animate-pulse'
      }`}
    >
      <div hidden={!message}>
        <Message message={message} />
      </div>
      <Players
        myUserId={me.userId}
        players={users.filter((u) => u.status !== Status.LEAVED)}
        shown={poker.state === 'EVERYONE_CHOSEN'}
      />
      <Button
        text="Reset"
        onClick={() => {
          setOpenResetDialog(true);
        }}
        glow={poker.state === 'EVERYONE_CHOSEN'}
      />
      <ModalDialog
        message="全員カードを未選択の状態にリセットします。よろしいですか？"
        onClickOK={onReset}
        open={openResetDialog}
        setOpen={setOpenResetDialog}
      />
    </div>
  );
};

const Container: FC = () => {
  const [me] = useRecoilState(myState);
  const [users] = useRecoilState(usersState);
  const poker = useRecoilValue(pokerRecoilState);
  const [openResetDialog, setOpenResetDialog] = useState(false);
  const setApp = useSetRecoilState(appState);

  const refresh = async () => {
    setApp((app) => ({ ...app, loading: true }));
    try {
      await mutateRefreshTable(me.roomId, me.userId);
    } catch (e) {
      setApp((app) => ({
        ...app,
        loading: false,
        errorMessage: NETWORK_ERROR,
      }));
    }
  };

  return (
    <Component
      me={me}
      users={users}
      poker={poker}
      onReset={() => refresh()}
      openResetDialog={openResetDialog}
      setOpenResetDialog={setOpenResetDialog}
      pokerState={poker.state}
    />
  );
};

export default Container;
