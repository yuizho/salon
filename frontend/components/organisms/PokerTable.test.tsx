import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { Status } from '../../graphql/schema';
import PokerTable from './PokerTable';

describe('PokerTable', () => {
  test('snapshot', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <PokerTable />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
