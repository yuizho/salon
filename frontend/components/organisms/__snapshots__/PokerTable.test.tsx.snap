// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokerTable snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col space-y-6 border rounded p-8 animate-pulse"
  >
    <div
      class="border-t border-b border-slate-400 text-slate-600 px-4 py-3"
      role="alert"
    >
      <p
        class="text-sm"
      >
        ã€€
      </p>
    </div>
    <div
      class="
  flex flex-wrap gap-2
  "
    />
    <div
      class="relative w-full"
    >
      <button
        class="
      relative
      py-2 px-4
      font-semibold
      border-slate-400
      text-slate-600
      border
      rounded
      shadow-md
      bg-white
      w-full
      hover:border-2
      active:shadow-none
      active:border
      active:border-slate-300
    "
        type="button"
      >
        Reset
      </button>
    </div>
  </div>
</DocumentFragment>
`;
